custom/icds_reports/reports/governance_apis.py

```diff
@@ -121,3 +121,47 @@ def base_data(row_data):
 def get_state_names():
     return list(AwcLocation.objects.filter(aggregation_level=AggregationLevels.STATE, state_is_test=0
                                            ).values('state_site_code', 'state_name'))
+
+
+@icds_quickcache(['length', 'year', 'month', 'order', 'query_filters'], timeout=30 * 60)
+def get_cbe_data(length, year, month, order, query_filters):
+    data = AggGovernanceDashboard.objects.filter(
+        month=date(year, month, 1),
+        **query_filters
+    ).order_by(*order).annotate(
+        cbe_conducted_1=Case(
+            When(cbe_date_1__isnull=False, then=Value('yes')),
+            default=Value('no'),
+            output_field=TextField(),
+        ),
+        cbe_conducted_2=Case(
+            When(cbe_date_2__isnull=False, then=Value('yes')),
+            default=Value('no'),
+            output_field=TextField(),
+        ),
+    ).values(
+        'awc_id',
+        'awc_code',
+        'cbe_conducted_1',
+        'cbe_type_1',
+        'cbe_date_1',
+        'num_target_beneficiaries_1',
+        'num_other_beneficiaries_1',
+        'cbe_conducted_2',
+        'cbe_type_2',
+        'cbe_date_2',
+        'num_target_beneficiaries_2',
+        'num_other_beneficiaries_2',
+    )
+
+    def get_value_or_data_not_entered(value):
+        if value is None:
+            return DATA_NOT_ENTERED
+        return value
+
+    def base_data(row_data):
```

Comment:
As an added bonus, if we got rid of all the formatting I flagged above we could avoid having to loop over the entries here, which would probably make this more performant

Reply:
removed the formatting.